### DISCUSSION TITLE
Making FromDigits more like StringContext; FromDigits literal suffixes

### DISCUSSION THREAD

**Discussion by s5bug** [scala/scala3#14501] [General Discussion]
Right now, `FromDigits` doesn't have any way to change the type of a literal without ascribing it. Ideally, one could get custom literals typed correctly via suffixes, such as `1_234bi` for a way to acquire `BigInt`s using underscores, and so on.

**Comment by bishabosha** [comment]
suffixes could work, except that identifiers would be limited by how they could be ambiguous when mixed with hexadecimal digits, I considered once using some kind of separator, like `0x7f'b` for a byte, which I implemented in [this commit](https://github.com/bishabosha/dotty/commit/a5d29acaa10fe161feb332ae53f6f7ca64f852db)

**Reply by som-snytt** [reply]
interesting idea!

**Comment by soronpo** [comment]
Since we are dealing with terms, perhaps we can use `#` in prefix position:
`b#0x7f` 
`bi#1234_5678_8910_1213_1415_16_17`
`ip#127.0.0.1`
So maybe we do not need `FromDigits` at all, but just a special case for string interpolation syntax to make it simpler for numerical expressions. This way `bi#1234_5678_8910_1213_1415_16_17` will be equivalent to `bi"1234_5678_8910_1213_1415_16_17"`.

**Reply by soronpo** [reply]
@odersky I'm interested in your thoughts about this proposal.
